usaspending_api/etl/swap_in_new_table_strategies.py

```diff
@@ -0,0 +1,944 @@
+
+            self._validate_tables(cursor)
+
+            self.logger.info(
+                f"Starting swap procedures for table {self.temp_schema_name}.{self.temp_table_name} "
+                f"into {self.curr_schema_name}.{self.curr_table_name}"
+            )
+
+            # Check if we're dealing with a partitioned table, or a partition of a partitioned table
+            self.is_temp_table_partitioned = is_table_partitioned(
+                table=f"{self.temp_schema_name}.{self.temp_table_name}", cursor=cursor
+            )
+            self.temp_table_parent_partitioned_table = get_parent_partitioned_table(
+                table=f"{self.temp_schema_name}.{self.temp_table_name}", cursor=cursor
+            )
+            self.is_curr_table_partitioned = is_table_partitioned(
+                table=f"{self.curr_schema_name}.{self.curr_table_name}", cursor=cursor
+            )
+            self.curr_table_parent_partitioned_table = get_parent_partitioned_table(
+                table=f"{self.curr_schema_name}.{self.curr_table_name}", cursor=cursor
+            )
+
+            # Make sure old and new are like-like in all ways
+            self._validate_state_of_tables(cursor)
+
+            # Get constraints and indexes named correctly before the table is renamed
+            self._swap_constraints_sql(cursor)
+            self._swap_index_sql(cursor)
+
+            # New temp views are setup to point to the temp table, and will swap-in with the table
+            self._create_new_views(cursor)
+
+            # Prep stats and privs on the staged temp table before swapping (renaming)
+            self._analyze_temp_table(cursor)
+            self._grant_privs_on_temp_table(cursor)
+
+            # Do the swaps via renames
+            self._swap_dependent_views(cursor)
+            self._swap_table(cursor)
+
+            if not self._options["keep_old_data"]:
+                self._drop_old_views(cursor)
+                if self.curr_table_parent_partitioned_table:
+                    self.logger.info(
+                        f"'{self.curr_table_name}' is a partition of '{self.curr_table_parent_partitioned_table}' and "
+                        f"will not be dropped. Swapped-out partitions are not deleted after the swap due to the risk "
+                        f"of their live parent partitioned table still receiving traffic while having one of its "
+                        f"partitions deleted. If '{self.curr_table_parent_partitioned_table}' is also being swapped, "
+                        f"then this partition will be deleted when it is deleted. Otherwise delete "
+                        f"'{self.curr_table_name}{self.old_suffix}' manually."
+                    )
+                else:
+                    self._drop_old_table_sql(cursor)
+
+            # If `delta_table_load_version_key` exists in the table_spec, this table could potentially be loaded incrementally,
```

Comment:
As this is included in both strategies, I think it would make more sense to be moved into the main command to remove code duplication.

Reply:
(or into the base class)

Reply:
Be warned moving this to a shared class couples these classes together. I made the change you requested